flow/src/benchmarking.rs

```diff
@@ -1,16 +1,148 @@
+	let token_symbol: Vec<u8> = vec![0; 5];
+	let token_name: Vec<u8> = vec![0; 32];
+	Flow::<T>::create_campaign(
+		RawOrigin::Signed(caller.clone()).into(),
+		org_id,
+		caller.clone(),
+		name,
+		T::MinContribution::get(),
+		T::MinContribution::get(),
+		frame_system::Pallet::<T>::block_number() + 200_u32.into(),
+		Default::default(),
+		Default::default(),
+		cid,
+		token_name,
+		token_symbol
+	)?;
+	let nonce = Nonce::<T>::get() - 1u128;
+	Ok(T::Hashing::hash_of(&nonce))
+}
+
 
 benchmarks! {
 
-	simple_benchmark {}: {}
-    // <T as orml_tokens::Config>::Currency::set_balance(GameCurrencyId, &creator, 1000);
+	create_campaign {
+		let b in 0 .. T::MaxCampaignsPerBlock::get()-1;  // already created campaigns at current block
+
+		let caller: T::AccountId = whitelisted_caller();
+		fund_account::<T>(&caller, None)?;
+		let org_id = T::Control::create_org(caller.clone().into())?;
+		let treasury_id = T::Control::org_treasury_account(&org_id);
+		fund_account::<T>(&treasury_id, None)?;
+		for i in 0..b {
+			create_campaign_call::<T>(caller.clone(), org_id.clone())?;
+		}
+		let count_before = CampaignsCount::<T>::get();
+
+	}: _(
+		RawOrigin::Signed(caller.clone()),
+		org_id,
+		caller.clone(),
+		vec![0; T::MaxNameLength::get() as usize],
+		T::MinContribution::get(),
+		T::MinContribution::get(),
+		200_u32.into(),
+		Default::default(),
+		Default::default(),
+		vec![0; T::MaxNameLength::get() as usize],
+		vec![0; 5],
+		vec![0; 32]
+	)
+	verify {
+		assert!(CampaignsCount::<T>::get() == count_before + 1);
+	}
+
+	update_state {
+		let caller: T::AccountId = whitelisted_caller();
+		fund_account::<T>(&caller, None)?;
+		let org_id = T::Control::create_org(caller.clone().into())?;
+		let treasury_id = T::Control::org_treasury_account(&org_id);
+		fund_account::<T>(&treasury_id, None)?;
+		let campaign_id = create_campaign_call::<T>(caller.clone(), org_id.clone())?;
+		let new_state = FlowState::Paused;
+	}: _(
+		RawOrigin::Signed(caller.clone()),
+		campaign_id,
+		new_state.clone()
+	)
+	verify {
+		assert!(CampaignState::<T>::get(&campaign_id) == new_state);
+	}
+
+	contribute {
+		let owner: T::AccountId = whitelisted_caller();
+		let contributor: T::AccountId = account("contributor", 0, 0);
+		fund_accounts::<T>(&vec![owner.clone(), contributor.clone()], None)?;
+		let org_id = T::Control::create_org(owner.clone().into())?;
+		let treasury_id = T::Control::org_treasury_account(&org_id);
+		fund_account::<T>(&treasury_id, None)?;
+		let campaign_id = create_campaign_call::<T>(owner, org_id)?;
+	}: _(
+		RawOrigin::Signed(contributor.clone()),
+		campaign_id.clone(),
+		T::MinContribution::get()
+	)
+	verify {
+		assert!(CampaignContribution::<T>::contains_key((&campaign_id, &contributor)));
+	}
+
+	on_initialize {
+		let b in 0 .. T::MaxContributorsProcessing::get();  // number of contributions in current block
+		let c in 0 .. T::MaxCampaignsPerBlock::get();  // number of campaigns in current block
```

Comment:
As a general suggestion - descriptive variable names and maybe more comments:
b => contributors
c => campaigns
frame_system::Pallet::<T>::block_number() + 200_u32.into() => expiry